src/client/datascience/notebookStorage/vscNotebookModel.ts

```diff
@@ -164,33 +163,6 @@ export class VSCodeNotebookModel extends BaseNotebookModel {
                 json.metadata = metadata;
             }
         }
-        // if (this.document && !this.isTrusted && Array.isArray(json.cells)) {
-        if (Array.isArray(json.cells)) {
-            // The output can contain custom metadata, we need to remove that.
-            json.cells = json.cells.map((cell) => {
-                const metadata = { ...cell.metadata };
-                // eslint-disable-next-line @typescript-eslint/no-explicit-any
-                const outputs: nbformat.IOutput[] = Array.isArray(cell.outputs) ? (cell.outputs as any) : [];
-                outputs.forEach((output: nbformat.IOutput) => {
```

Comment:
Remove all of this... better strongly typed metadata stored in the output & cells.
& the custom metadata we store will not get added to the Jupyter cells.